### DISCUSSION TITLE
How to create soft shadow

### DISCUSSION THREAD

**Discussion by pseandersson** [google/filament#3667] [Q&A]
Hi,
I have been using filament for a while, and now I want to achieve a soft shadow effect on a shadow capture plane. I have tried downsample the shadow map but the quality of the downsample map is too bad.

Does filament support softshadow/arealight and how do I achieve it with the android apis?

Kind regards
P

**Comment by romainguy** [comment]
VSM shadows will allow for softer shadows once we add the blur option, which is planned. @bejado 

Otherwise you'd have to generate your own projected shadow using render targets for instance.

**Reply by pseandersson** [reply]
I have tried using VSM shadows, but unfortunatly they did not work with my shadow catcher. Perhaps I set up something wrong in my ModelView or are they not supported right now? I wonder also if you have a roadmap for VSM which one can look on?

**Reply by bejado** [reply]
VSM doesn't currently support shadow catchers, but it's on our backlog of features to add.

**Reply by bejado** [reply]
The shadow catcher fix has been merged into main.

**Reply by pseandersson** [reply]
Great! Thank you!
